units/src/amount/signed.rs

```diff
@@ -249,7 +248,7 @@ impl SignedAmount {
     pub const fn checked_add(self, rhs: SignedAmount) -> Option<SignedAmount> {
         // No `map()` in const context.
         match self.0.checked_add(rhs.0) {
-            Some(res) => Some(SignedAmount(res)),
+            Some(res) => SignedAmount(res).check_min_max(),
```

Comment:
I don't think this makes sense to check twice.  There's a way to do this where we don't need to check `self.0.checked_add(rhs.0)` as well as `.check_min_max()`.  right?  same for the other operations.

Reply:
In other words, do a checked add, and if the results exceed `MAX_MONEY`, then `NONE` else `Some(signed_amount)`

Reply:
I don't understand what you mean by check twice.  The first `checked_add()` will only fail if it overflows, it has nothing to do with our `MAX` invariant.  There is only one check for `MIN` and `MAX`, the `.check_min_max()` private function.  I made it this way so that the check for all the public functions are not repeated and are done by one function which returns the Option.

Reply:
this could be verified with an assertion added to a test.  My point is that `checked_add` should check that adding two numbers is not greater than `MAX_MONEY`, then if is not greater than `MAX_MONEY` you don't need to check that it overflows do you?

Reply:
This change looks correct to me.

Reply:
It's not uncorrect to check both that it is not greater than MAX_MONEY and not greater than `i64::MAX` just unnecessary.

Reply:
\me says a little prayer "please lord give me patience"